File: app/src/main/java/ru/geekbrains/notes/GlobalVariables.java
package ru.geekbrains.notes;

import android.app.Application;

import java.util.ArrayList;
import java.util.List;

import ru.geekbrains.notes.note.Note;

public class GlobalVariables extends Application {

    private List<Note> notes;

    private int currentNote;
    private int textSizeId;
    private int sortTypeId;
    private boolean viewNoteFragmentState;

    public boolean isViewNoteFragmentState() {
        return viewNoteFragmentState;
    }

    public void setViewNoteFragmentState(boolean viewNoteFragmentState) {
        this.viewNoteFragmentState = viewNoteFragmentState;
    }



    public int getTextSizeId() {
        return textSizeId;
    }

    public void setTextSizeId(int textSizeId) {
        this.textSizeId = textSizeId;
    }

    public int getSortTypeId() {
        return sortTypeId;
    }

    public void setSortTypeId(int sortTypeId) {
        this.sortTypeId = sortTypeId;
    }

    public int getCurrentNote() {
        Note note = getNoteByNoteId(currentNote);
        if (note.getID() != -1)
            return currentNote;
        else
            return notes.size() - 1;
    }

    public void setCurrentNote(int currentNote) {
        this.currentNote = currentNote;
    }

    public List<Note> getNotes() {
        return notes;
    }

    public List<Note> getNotesWithText(String query) {
        List<Note> result = new ArrayList<>();
        for (int i = 0; i < notes.size(); i++) {
            if (notes.get(i).getValue().toUpperCase().contains(query.toUpperCase()))
                result.add(notes.get(i));
        }
        return result;
    }

    public void setNotes(List<Note> notes) {
        this.notes = notes;
    }

    public Note getNoteByNoteId(int noteId){
        for (int i = 0; i < notes.size(); i++) {
            if (notes.get(i).getID() == noteId) {
                return notes.get(i);
            }
        }
        return new Note();
    }

    public int getScrollPositionByNoteId(int noteId){
        for (int i = 0; i < notes.size(); i++) {
            if (notes.get(i).getID() == noteId) {
                return i;
            }
        }
        return 0;
    }

    public int getNewId(){
        int newId = 0;
        if (notes.size() > 0){
            newId = notes.get(0).getID();
            for (int i = 1; i < notes.size(); i++) {
                if (notes.get(i).getID() > newId) {
                    newId = notes.get(i).getID();
                }
            }
            newId++;
        }
        return newId;
    }

    public void setNoteById(int noteId, Note note){
        for (int i = 0; i < notes.size(); i++) {
            if (notes.get(i).getID() == noteId) {
                notes.set(i, note);
            }
        }
    }

}

File: app/src/main/java/ru/geekbrains/notes/GlobalVariables.kt
package ru.geekbrains.notes

import android.app.Application
import ru.geekbrains.notes.note.Note
import java.util.*

class GlobalVariables : Application() {
    private var notes: MutableList<Note>? = null
    private var notesCloud: MutableList<Note>? = null
    var currentNote = 0
        get() {
            val note = getNoteByNoteId(field)
            return if (note.iD != -1) field else notes!!.size - 1
        }
    var isViewNoteFragmentState = false
    var settings: Settings? = null
    fun getNotes(): List<Note>? {
        return notes
    }

    fun getNotesWithText(query: String): List<Note> {
        val result: MutableList<Note> = ArrayList()
        for (i in notes!!.indices) {
            if (notes!![i].value!!.toUpperCase().contains(query.toUpperCase())) result.add(notes!![i])
        }
        return result
    }

    fun setNotes(notes: MutableList<Note>?) {
        this.notes = notes
    }

    fun setNotesCloud(notes: MutableList<Note>?) {
        notesCloud = notes
    }

    fun getNotesCloud(): List<Note>? {
        return notesCloud
    }

    fun getScrollPositionByNoteId(noteId: Int): Int {
        for (i in notes!!.indices) {
            if (notes!![i].iD == noteId) {
                return i
            }
        }
        return 0
    }

    val newId: Int
        get() {
            var newId = 0
            if (notes!!.size > 0) {
                newId = notes!![0].iD
                for (i in 1 until notes!!.size) {
                    if (notes!![i].iD > newId) {
                        newId = notes!![i].iD
                    }
                }
                newId++
            }
            return newId
        }

    fun setNoteById(noteId: Int, note: Note) {
        for (i in notes!!.indices) {
            if (notes!![i].iD == noteId) {
                notes!![i] = note
            }
        }
    }

    fun getNoteByNoteId(noteId: Int): Note {
        for (i in notes!!.indices) {
            if (notes!![i].iD == noteId) {
                return notes!![i]
            }
        }
        return Note()
    }

    fun updateNoteCloudById(noteId: Int, note: Note) {
        for (i in notesCloud!!.indices) {
            if (notesCloud!![i].iD == noteId) {
                notesCloud!![i] = note
            }
        }
    }

    fun getNoteCloudByNoteId(noteId: Int): Note {
        for (i in notesCloud!!.indices) {
            if (notesCloud!![i].iD == noteId) {
                return notesCloud!![i]
            }
        }
        return Note()
    }
}